src/app/authentication/login.ts

```diff
@@ -212,6 +275,48 @@ export const loginDeviceUpgradeWithPhone = async ({
     // a. create kratos account
     // b. and c. migrate account/user collection in mongo via kratos/registration webhook
 
+    // check if account is upgradeable
+    const accountConfig = getAccountCountries()
+
+    if (accountConfig.enableIpCheck) {
```

Comment:
I think these 2 conditionals `enableIpCheck` and `enablePhoneCheck` should not be present in the application layer. It's very procedural ins style.

Better would be to encapsulate the `isAllowedToLogin` logic in a (unit-tested) domain object that gets executed unconditionally.

Reply:
I refactored the code https://github.com/GaloyMoney/galoy/pull/2905/commits/1110713ff1c433dfdb0754583109072b12840450 but still kept it at the app level. the thing is this code is mostly dealing with network/services call here, so it's unclear to me how to best handle it if we were to move it into domain.